app/src/userpages/components/Avatar/CropAvatarDialog/index.jsx

```diff
@@ -0,0 +1,96 @@
+// @flow
+
+import React from 'react'
+import { I18n } from 'react-redux-i18n'
+import AvatarEditor from 'react-avatar-editor'
+
+import Dialog from '$shared/components/Dialog'
+import Slider from '$shared/components/Slider'
+
+import styles from './cropAvatarDialog.pcss'
+
+type Props = {
+    originalImage: string,
+    onClose: () => void,
+    onSave: (string) => void,
+}
+
+type State = {
+    sliderValue: number,
+}
+
+type Editor = {
+    getImage: () => HTMLCanvasElement,
+    getImageScaledToCanvas: () => HTMLCanvasElement,
+}
+
+class CropAvatarDialog extends React.Component<Props, State> {
+    state = {
+        sliderValue: 1,
+    }
+
+    editor: ?Editor = null
+
+    onSliderChange = (value: number) => {
+        this.setState({
+            sliderValue: value,
+        })
+    }
+
+    setCanvasRef = (editor: ?Editor) => {
```

Comment:
Maybe use `createRef` API: https://reactjs.org/docs/refs-and-the-dom.html#creating-refs